### PR TITLE
Add pruning interval for KerasPruningCallback

### PR DESCRIPTION
**Pull request by VladSkripniuk**
Address #1136.

### CODE REVIEW

File: optuna/integration/keras.py [hunk 1]
```diff
@@ -31,21 +31,29 @@ class KerasPruningCallback(Callback):
             An evaluation metric for pruning, e.g., ``val_loss`` and
             ``val_acc``. Please refer to `keras.Callback reference
             <https://keras.io/callbacks/#callback>`_ for further details.
-    """
+        interval:
+            Check if trial should be pruned every n-th epoch. By default `interval=1` and
+            pruning is performed after every epoch. Increase `interval` to run several
```
Comment (hvy) [inline optuna/integration/keras.py:36]:
```suggestion
            Check if trial should be pruned every n-th epoch. By default ``interval=1`` and
            pruning is performed after every epoch. Increase ``interval`` to run several
```

File: tests/integration_tests/test_keras.py [hunk 2]
```diff
@@ -9,8 +9,9 @@
 from optuna.testing.integration import DeterministicPruner
 
 
-def test_keras_pruning_callback():
-    # type: () -> None
+@pytest.mark.parametrize("interval, epochs", [(1, 1), (2, 1), (2, 2)])
+def test_keras_pruning_callback(interval, epochs):
```
Comment (keisuke-umezawa) [inline tests/integration_tests/test_keras.py:13]: Just a nit, and we can fix it in another follow up PR.

For me, it is easier to understand what they are testing if we divide `test_keras_pruning_callback` and `test_keras_pruning_callback_with_interval`.

c.f. 
https://github.com/optuna/optuna/pull/1154/files#diff-58bac36a299f885bfb5a18600c4b6cb1R110

File: tests/integration_tests/test_keras.py [hunk 3]
```diff
@@ -22,16 +23,19 @@ def objective(trial):
...
 
         return 1.0
 
     study = optuna.create_study(pruner=DeterministicPruner(True))
     study.optimize(objective, n_trials=1)
-    assert study.trials[0].state == optuna.trial.TrialState.PRUNED
+    if interval <= epochs:
+        assert study.trials[0].state == optuna.trial.TrialState.PRUNED
+    else:
+        assert study.trials[0].state == optuna.trial.TrialState.COMPLETE
```
Comment (keisuke-umezawa) [inline tests/integration_tests/test_keras.py:38]: Just a nit, and we can fix it in another follow up PR.

How about adding this line below?
```
assert study.trials[0].value == 1.0
```

Comment (VladSkripniuk) [reply tests/integration_tests/test_keras.py:38]: Right, I'll open a follow up PR once this PR is merged.

**Comment by hvy** [comment]
@toshihikoyanase sorry for pinging but did you have any comments on including the 0-th step or any other details?

**Comment by toshihikoyanase** [comment]
@hvy Thank you for reminding me.

I notice that users can control pruning interval in two ways if we add `interval` to pruning callbacks. For example, [`MedianPruner`](https://optuna.readthedocs.io/en/stable/reference/pruners.html#optuna.pruners.MedianPruner) has `n_warmup_steps` and `interval_steps` to control the steps to check the intermediate values. Users can configure the pruning interval by both `interval` of `KerasPruningCallback` and `interval_steps` of `MedianPruner`.

I'm thinking if we can unify these to make the interface simple.

This is just an example, but how about adding a method to check the pruning interval to `BasePruner` (e.g., the `is_target_step` method of the following snippet)?

```python

    def on_epoch_end(self, epoch, logs=None):
        # type: (int, Dict[str, float]) -> None

        if not self._trial.study.pruner.is_target_step(epoch):
            return

```

IMO, users usually use the `interval` of `KerasPruningCallback` to stop reporting intermediate values which are not referred by pruners. If so, it seems to be natural to calculate pruning interval in pruners.

What do you think?

**Comment by toshihikoyanase** [comment]
@hvy
I think it takes time to design/implement `Pruner.is_target_step`, so I'd like to create a new PR if I work on it. Please merge this PR.
Thank you for your kind consideration

**Comment by hvy** [comment]
Thanks @VladSkripniuk for the long running PR, and @toshihikoyanase for you careful insight. For reasons stated above, I've merged this PR.

**Pull request merged**